src/api/init.ts

```diff
@@ -128,6 +128,16 @@ export async function startApiServer(opts: {
     });
     app.use(promMiddleware);
   }
+  // adding stacks-api-version in header
+  app.use((_, res, next) => {
+    try {
+      const [branch, commit, tag] = fs.readFileSync('.git-info', 'utf-8').split('\n');
```

Comment:
Should definitely avoid reading this file on _every_ request. These variables can be read and stored in a variable outside of this function.